### PR TITLE
fix(query): global contain stage and udf priv

### PR DESCRIPTION
**Pull request by TCeason**
I hereby agree to the terms of the CLA available at: https://docs.databend.com/dev/policies/cla/

## Summary

1. Global contains all objects
2. ALL privilege should not contain Ownership privilege.

Fixes #[Link the issue here]

## Tests

- [ ] Unit Test
- [x] Logic Test
- [ ] Benchmark Test
- [ ] No Test  - _Explain why_

## Type of change

- [x] Bug Fix (non-breaking change which fixes an issue)
- [ ] New Feature (non-breaking change which adds functionality)
- [ ] Breaking Change (fix or feature that could cause existing functionality not to work as expected)
- [ ] Documentation Update
- [ ] Refactoring
- [ ] Performance Improvement
- [ ] Other (please describe):

- - -
This change is [Reviewable](https://reviewable.io/reviews/datafuselabs/databend/14344)

### CODE REVIEW

File: src/query/sql/src/planner/binder/ddl/account.rs [hunk 1]
```diff
@@ -55,7 +55,7 @@ impl Binder {
                 // ALL PRIVILEGES have different available privileges set on different grant objects
                 // Now in this case all is always true.
                 let grant_object = self.convert_to_grant_object(level).await?;
-                let priv_types = grant_object.available_privileges();
+                let priv_types = grant_object.available_privileges_without_ownership();
```
Comment (flaneur2020) [inline src/query/sql/src/planner/binder/ddl/account.rs:58]: how about just `available_privileges` ?

ownership in our design is not considered as a real privilege. we can put a description about this in the comment of this function.

File: src/meta/app/src/principal/user_privilege.rs [hunk 2]
```diff
@@ -165,15 +179,29 @@ impl UserPrivilegeSet {
...
     pub fn available_privileges_on_stage() -> Self {
-        make_bitflags!(UserPrivilegeType::{  Read | Write | Ownership }).into()
+        (UserPrivilegeSet::privileges_on_stage_without_ownership().privileges
+            | make_bitflags!(UserPrivilegeType::{  Ownership }))
+        .into()
+    }
+
+    pub fn privileges_on_stage_without_ownership() -> Self {
+        make_bitflags!(UserPrivilegeType::{  Read | Write }).into()
+    }
```
Comment (flaneur2020) [inline src/meta/app/src/principal/user_privilege.rs:195]: if we need seperate the cases about containing ownership or not, we can consider merge 
`privileges_on_stage_without_ownership` and `privileges_on_stage` functions with a single one like `privileges_on_stage(contain_ownership: bool)`.

Comment (TCeason) [reply src/meta/app/src/principal/user_privilege.rs:180]: Cool.

File: src/meta/app/src/principal/user_grant.rs [hunk 3]
```diff
@@ -175,8 +179,8 @@ impl GrantEntry {
         &self.object == object
     }
 
-    pub fn has_all_available_privileges(&self) -> bool {
-        let all_available_privileges = self.object.available_privileges();
+    pub fn has_all_available_privileges_without_ownership(&self) -> bool {
```
Comment (flaneur2020) [inline src/meta/app/src/principal/user_grant.rs:182]: I feel calling `has_all_available_privileges()` is ok, in the current design, having an ownership is not considered as having a privilege.

Comment (TCeason) [reply src/meta/app/src/principal/user_grant.rs:182]: ok. Done

**Review by flaneur2020 — APPROVED** [review]

**Comment by TCeason** [comment]
Please merge this pr. @BohuTANG

**Pull request merged**